### PR TITLE
Adding a clone_dataset operator

### PR DESCRIPTION
**Pull request by brimoor**
Adds a `@voxel51/utils/clone_dataset` operator

### CODE REVIEW

File: plugins/utils/README.md [hunk 1]
```diff
@@ -116,6 +116,23 @@ dataset.name = new_name
...
+### clone_dataset
+
+You can use this operator to clone a dataset or your current view in the App.
+
+This operator is essentially a wrapper around the
+[clone()](https://docs.voxel51.com/api/fiftyone.core.dataset.html#fiftyone.core.dataset.Dataset.clone)
+method:
+
+```py
+new_dataset = dataset.clone(new_name)
```
Comment (minhtuev) [inline plugins/utils/README.md:128]: nice!

File: plugins/utils/__init__.py [hunk 2]
```diff
@@ -1077,6 +1077,135 @@ def execute(self, ctx):
...
+    def resolve_input(self, ctx):
+        inputs = types.Object()
+
+        _get_clone_dataset_inputs(ctx, inputs)
+
+        return types.Property(inputs, view=types.View(label="Clone dataset"))
+
+    def execute(self, ctx):
+        name = ctx.params["name"]
+        new_name = ctx.params["new_name"]
```
Comment (minhtuev) [inline plugins/utils/__init__.py:1100]: do we need to check if the new name is invalid and throw an error if that is true before executing?

**Review by minhtuev — APPROVED** [review]
A small comment, otherwise LGTM!

File: plugins/utils/__init__.py (hunk 2 above)
Comment (brimoor) [reply plugins/utils/__init__.py:1100]: Validation of `new_name` is performed in `resolve_input()` to make sure the name isn't already taken.

If someone else created a `new_name` dataset between when `resolve_input()` and `execute()` are executed, then `clone()` will raise an informative error that informs the user of it.

**Pull request merged**